collectors/metrics/cmd/metrics-collector/main.go

```diff
@@ -362,21 +361,24 @@ func (o *Options) Run() error {
 			return worker.Reconfigure(*cfg)
 		})
 		handlers.Handle("/federate", serveLastMetrics(o.Logger, worker))
-		l, err := net.Listen("tcp", o.Listen)
-		if err != nil {
-			return fmt.Errorf("failed to listen: %w", err)
+		s := http.Server{
+			Addr:              o.Listen,
+			Handler:           handlers,
+			ReadHeaderTimeout: 1 * time.Second,
+			ReadTimeout:       5 * time.Second,
```

Comment:
Tbh, I copied this verbatim from obs API. I think the write timeout was bigger due to Loki stuff there. 
Over here, I guess it will be remote write request to Obs API.

But yeah I can make both the same if you want.

Reply:
I'm mostly worried about the read timeout being so low than the write timeout being too big. Maybe we could bump the read timeout to something like 15s and the read header one to 5s? 🤔

Reply:
Done!